### PR TITLE
[careful + breaking change] Removed Handlebars 3 support

### PR DESCRIPTION
**Pull request by matteofigus**
This is the last PR for fixing #252 - following #243 

* This will trigger a new minor version as it's a breaking change
* All the handlebars 3 stuff has been removed

### CODE REVIEW

**Comment by matteofigus** [comment]
I think this could be kind of ready if you want to start doing a review @antwhite / @mattiaerre?

File: src/components/oc-client/_package/src/oc-client.js [hunk 1]
```diff
@@ -302,41 +301,16 @@ var oc = oc || {};
...
-
-                  callback(null, html);
-                });
               });
-
-              // End of the Handlebars "migrate" part
-
             } else if(compiledViewInfo.type === 'jade'){
-              oc.require('jade', JADE_URL, function(jade){
+              oc.require('jade', JADE_URL, function(){
```
Comment (mattiaerre) [inline src/components/oc-client/_package/src/oc-client.js]: this is not related right? you just removed the parameter as not used by that function correct? a.k.a. "[the boy scout rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule)" 😄

File: src/components/oc-client/src/oc-client.js [hunk 2]
```diff
@@ -302,41 +301,16 @@ var oc = oc || {};
...
-
-                  callback(null, html);
-                });
               });
-
-              // End of the Handlebars "migrate" part
-
             } else if(compiledViewInfo.type === 'jade'){
-              oc.require('jade', JADE_URL, function(jade){
+              oc.require('jade', JADE_URL, function(){
```
Comment (mattiaerre) [inline src/components/oc-client/src/oc-client.js]: same as above right?

Comment (matteofigus) [inline src/components/oc-client/src/oc-client.js]: Yes to both :)

File: client/src/validator.js [hunk 3]
```diff
@@ -4,6 +4,20 @@ var settings = require('./settings');
 var _ = require('./utils/helpers');
 
 module.exports = {
+  validateComponent: function(template, options){
```
Comment (mattiaerre) [inline client/src/validator.js]: what about adding a unit test around this behaviour?

**Comment by mattiaerre** [comment]
not sure how it works w/ licenses but what about this?

https://github.com/opentable/oc/blob/handlebars4/src/components/oc-client/LICENSES#L80

File: test/front-end/render.js [hunk 4]
```diff
@@ -37,24 +37,14 @@ describe('oc-client : render', function(){
       var originalHb3, originalHb4;
```
Comment (mattiaerre) [inline test/front-end/render.js]: I think we do not need these variables anymore do we?

Comment (matteofigus) [inline test/front-end/render.js]: Ok I cleaned up this

File: client/src/validator.js (hunk 3 above)
Comment (matteofigus) [inline client/src/validator.js]: Done 👍

**Comment by antwhite** [comment]
LGTM

**Comment by matteofigus** [comment]
Thanks! So, @mattiaerre can you merge when you're happy about the tests? (I added some as requested).

File: test/unit/client-validator.js [hunk 5]
```diff
@@ -32,4 +32,43 @@ describe('client : validator', function(){
       });
     });
   });
+
+  describe('when validating component', function(){
```
Comment (mattiaerre) [inline test/unit/client-validator.js]: awesome 👍

**Comment by mattiaerre** [comment]
good job @matteofigus 
thanks,

// cc @antwhite
